### ISSUE TITLE
False positive f-string SyntaxError

### ISSUE THREAD

**Issue by youknowone** [astral-sh/ruff#7778]
```python
# x.py
f"{1:{{1}.pop()}{f'n'}}"
```

```sh
$ ruff x.py
error: Failed to parse x.py:1:16: f-string: single '}' is not allowed
x.py:1:16: E999 SyntaxError: f-string: single '}' is not allowed
Found 1 error.
$ ruff --version
ruff 0.0.292
```

**Comment by dhruvmanila** [comment]
Thanks for the bug report! It seems to be caused by the two `{{` inside the format spec part. It seems that the rules for escaping `{`/`}` are different in and out of a format spec.
